### ISSUE TITLE
training will not stop?

### ISSUE THREAD

**Issue by sed0724963** [tensorflow/models#5395]
1.Training will not stop?(that I was set how step will it stop,but it's not work)??
I used the model: "faster rcnn resnet101 coco",and using the sample config. Data it's my own data.

2.This config:I only change num_classes  and image size,and data from .
the step is the same 900000 steps.(ps:Does anyone know why it has two kinds of steps:900000 and 1200000?)
<img width="353" alt="2018-09-28 1 58 39" src="https://user-images.githubusercontent.com/38509051/46190569-ab9a7480-c326-11e8-828e-7386e55e9ce6.png">

3.When the mechine was training over than 900000 steps, it's still not stop?? why?(ps:it's also over than 1200000steps, weither 900000 or 1200000)

1.What is the top-level directory of the model you are using:research/object_detection
2.Have I written custom code:No
3.OS Platform and Distribution:win10
4.TensorFlow version:1.8
5.CUDA/cuDNN version:9.2&7.1
GPU model and memory:GTX 1080 8G

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
TensorFlow installed from
Bazel version
Exact command to reproduce

**Comment by Manish-rai21bit** [comment]
To your question on:
Point 2 - I believe the learning rates are updated to a much finer value after the 900,000 and 1,200,000 steps. As per my understanding, this will help in a pointed convergence and will avoid the gradient vector from shooting back and forth.
3. Did you give the training step as - 900,000 in the training pipeline? Can you show the training pipeline that you used?

**Comment by sed0724963** [comment]
1. I use this command line:python object_detection/train.py --logtostderr --pipeline_config_path=object_detection/data/faster_rcnn_resnet101_coco.config --train_dir=object_detection/data/      
, I don't know where it has set of training steps ??

2. That I used to use "ssd_mobilenet_v1_pets.config" to training and it will stop,so I 
misunderstanding that 900000 steps is the total training steps of "faster_rcnn_resnet101_coco.config" and think it will stop. Obviously they are different?

3. How can I do for let it(faster_rcnn_resnet101_coco.config) stop?
Additionally, where has any Instruction manual to let me to learning the sample config??

Thank you for your solution!

<img width="646" alt="2018-09-29 1 17 07" src="https://user-images.githubusercontent.com/38509051/46241346-32645580-c3ea-11e8-80da-26f48d0c6e81.png">

**Comment by Manish-rai21bit** [comment]
Hi,

I am using the latest version of the training module [model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py) which has the option to specify the training steps in the pipeline. This is how I specify the number of training steps in my pipeline.

`PIPELINE_CONFIG_PATH='training/faster_rcnn_resnet101_coco.config'
MODEL_DIR='./training/'
NUM_TRAIN_STEPS=90000
NUM_EVAL_STEPS=1500

python model_main.py \
    --pipeline_config_path=${PIPELINE_CONFIG_PATH}  \
    --model_dir=${MODEL_DIR} \
    --num_train_steps=${NUM_TRAIN_STEPS} \
    --num_eval_steps=${NUM_EVAL_STEPS} \
    --alsologtostderr
`

However, I see that ssd_mobilenetXXX.configs have the number of training steps as part of their config file but I do not see this option on the faster R-CNN config. I've not looked at the older "train.py" but the newer [model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py) does provide the option to pass the parameters (line 33).
I guess you should look at the "train.py" to see if you can pass in the parameters.

Best,

**Comment by sed0724963** [comment]
OK,understand !thank you very much.

**Comment by lukeb94** [comment]
> Hi,
> 
> I am using the latest version of the training module [model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py) which has the option to specify the training steps in the pipeline. This is how I specify the number of training steps in my pipeline.
> 
> `PIPELINE_CONFIG_PATH='training/faster_rcnn_resnet101_coco.config'
> MODEL_DIR='./training/'
> NUM_TRAIN_STEPS=90000
> NUM_EVAL_STEPS=1500
> 
> python model_main.py 
> --pipeline_config_path=${PIPELINE_CONFIG_PATH} 
> --model_dir=${MODEL_DIR} 
> --num_train_steps=${NUM_TRAIN_STEPS} 
> --num_eval_steps=${NUM_EVAL_STEPS} 
> --alsologtostderr
> `
> 
> However, I see that ssd_mobilenetXXX.configs have the number of training steps as part of their config file but I do not see this option on the faster R-CNN config. I've not looked at the older "train.py" but the newer [model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py) does provide the option to pass the parameters (line 33).
> I guess you should look at the "train.py" to see if you can pass in the parameters.
> 
> Best,

Hi all, I am still struggling with this issue. I have setup using faster_rcnn_resnet101_coco.config. I need to know how to set the training steps.

I found training_number_of_steps FLAG in the train.py, after including in the call I see no difference:

python /home/luke/Documents/FasterRCNN/models/research/object_detection/legacy/train.py 
--train_dir=/home/luke/Documents/FasterRCNN/Tensorflow_Object_detection_API-Custom_Faster_RCNN-master/CP 
--pipeline_config_path=/home/luke/Documents/FasterRCNN/Tensorflow_Object_detection_API-Custom_Faster_RCNN-master/faster_rcnn_resnet101_coco.config 
--training_number_of_steps=1

Please let me know if anyone has a fix, I am trying to generate the .pb file. Many thanks

**Comment by DevLob-zz** [comment]
goto to and add this node train_config: { num_steps: 5000
